junit-runtime/src/main/scala/org/scalajs/junit/JUnitExecuteTest.scala

```diff
@@ -64,116 +64,101 @@ final class JUnitExecuteTest(taskDef: TaskDef, runner: JUnitBaseRunner,
       method: JUnitMethodMetadata) = {
     val jUnitMetadata = classMetadata.metadata()
     val testClassInstance = classMetadata.newInstance()
+    val methodName = method.name
+    val testAnnotation = method.getTestAnnotation.get
 
     val t0 = System.nanoTime
-    def timeInSeconds() = (System.nanoTime - t0).toDouble / 1000000000
+    def timeInSeconds = (System.nanoTime - t0).toDouble / 1000000000
 
-    val beforeMethodsAssertionFaild = {
+    def executeTestMethods(): Unit = {
+      val expectedException = testAnnotation.expected
       try {
-        for (method <- jUnitMetadata.beforeMethod)
-          classMetadata.invoke(testClassInstance, method.name)
-        false
-      } catch {
-        case ex: AssumptionViolatedException =>
-          logFormattedInfo(method.name, "started")
-          logAssertionWarning(method.name, ex, timeInSeconds())
-          true
-
-        case ex: internal.AssumptionViolatedException =>
-          logFormattedInfo(method.name, "started")
-          logAssertionWarning(method.name, ex, timeInSeconds())
-          true
-      }
-    }
+        if (runner.runSettings.verbose)
+          logFormattedInfo(methodName, "started")
 
-    if (!beforeMethodsAssertionFaild) {
-      val testAnnotation = method.getTestAnnotation.get
-      val testMethodFailed = {
-        try {
-          classMetadata.invoke(testClassInstance, method.name)
-          executedWithoutExceptions(method.name, testAnnotation, timeInSeconds())
-          false
-        } catch {
-          case ex: Throwable =>
-            executedWithExceptions(method.name, testAnnotation, timeInSeconds(), ex)
-            true
+        classMetadata.invoke(testClassInstance, methodName)
+
+        if (expectedException == classOf[org.junit.Test.None]) {
+          taskPassed(methodName)
+          executeAfterMethods()
```

Comment:
This is wrong (although seems to reproduce the old behavior). `@AfterClass` methods always need to be run: http://junit.sourceforge.net/javadoc/org/junit/AfterClass.html